### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by dmitruz**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = JSON.parse(JSON.stringify(state));
+  const result = [stateCopy];
```
Comment (etojeDenys) [inline src/transformStateWithClones.js:11]:
```suggestion
  const result = [];
```

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,40 @@
...
+
+  for (const action of actions) {
+    // Creating new copy for each iteration
+    const newState = JSON.parse(JSON.stringify(result[result.length - 1]));
```
Comment (etojeDenys) [inline src/transformStateWithClones.js:15]:
```suggestion

```

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,40 @@
...
+      }
+      case 'clear':
+        // next create new epty state
+        Object.keys(newState).forEach(key => delete newState[key]);
+        break;
+      default:
+        // handle an error
+        throw new Error(`Invalid action type: ${action.type}`);
+    }
+    result.push(newState);
```
Comment (etojeDenys) [inline src/transformStateWithClones.js:39]:
```suggestion
    result.push({...newState});
```

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,40 @@
...
+  }
+  result.shift();
```
Comment (etojeDenys) [inline src/transformStateWithClones.js:41]:
```suggestion
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]
we can simplify it

File: src/transformStateWithClones.js (hunk 4 above)
Comment (etojeDenys) [reply src/transformStateWithClones.js:41]: feel free to ask me for some help in the chat

**Comment by dmitruz** [comment]
const newState = JSON.parse(JSON.stringify(result[result.length - 1]))   If I delete this line it is not working. Than I have to delete NewState word everywhere?

**Review by etojeDenys — APPROVED** [review]
looks good to me

**Pull request closed without merging**